corehq/apps/app_manager/util.py

```diff
@@ -431,9 +431,13 @@ def get_session_schema(form):
     from corehq.apps.app_manager.suite_xml.sections.entries import EntriesHelper
     structure = {}
     datums = EntriesHelper(form.get_app()).get_datums_meta_for_form_generic(form)
-    datums = [d for d in datums if not d.is_new_case_id and d.case_type and d.datum.id == 'case_id']
+    datums = [
+        d for d in datums
+        if not d.is_new_case_id and d.case_type and d.requires_selection
+    ]
     if len(datums):
-        structure['case_id'] = {
+        session_var = datums[-1].datum.id
+        structure[session_var] = {
```

Comment:
Will `session_var` ever be anything other than `"case_id"`? I wonder if that could break case hashtags in Vellum since Vellum basically has one case hashtag prefix `#case/` which means `instance('casedb')/cases/case[@case_id = instance('commcaresession')/session/data/case_id]/`? On looking I'm not sure if any of that is hard-coded in Vellum, but I seem to remember that some of it had been at one time.

[Documentation for the caseDB schema](https://github.com/dimagi/Vellum/blob/master/src/datasources.js) can be found in Vellum. There is a similar comment in HQ in the [docstring of `get_form_data_schema`](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/views/formdesigner.py#L146)